kn/fotos/media/fotos.js

```diff
@@ -92,40 +96,52 @@
   };
 
   KNF.prototype.fetch_fotos = function() {
-    var old_path = this.path;
+    var path = this.path;
+    if (this.fotos[path] === null) {
+      // fetch in progress
+      return;
+    }
+    this.fotos[path] = null;
     this.api({action: 'list',
-              path: this.path},
+              path: path},
       function(data) {
-        if(old_path != this.path) return;
         if(data.error) {
           $('#fotos').text(data.error);
           return;
         }
 
-        this.fotos = {};
-        var prev = null;
-        $.each(data.children, function(i, c) {
-          c.thumbnail = this.cache_url('thumb', c.path);
-          c.thumbnail2x = this.cache_url('thumb2x', c.path);
-          if (c.type == 'foto') {
-            c.full = this.cache_url('full', c.path);
-            c.large = this.cache_url('large', c.path);
-            c.large2x = this.cache_url('large2x', c.path);
-          }
-          this.fotos[c.name] = c;
-          if (prev !== null) {
-            prev.next = c;
-            c.prev = prev;
-          }
-          prev = c;
-        }.bind(this));
+        this.read_fotos(path, data);
+
         this.display_fotos();
-        if (this.get_hash() && this.foto === null) {
-          this.change_foto(this.fotos[this.get_hash()]);
+        if (path == this.path && this.get_hash() && this.foto === null) {
```

Comment:
Goed punt. Ik verwacht van niet, maar het is nu gefixt. Het is beter om altijd `===` te gebruiken.